### PR TITLE
Feature/adhithya christen/integrate mongo db

### LINKED ISSUE #6: Integrate NodeJS server with MongoDB
Note: Once you are assigned to this task, please send me your email ids so that I can add you to the mongoDB cluster we have created. 

## What do we need?
- First step to setting up our backend is to integrate the server with our database in MongoDB. 
- Once done, we should be smoothly able to store / retrieve documents to and from our database
- Use package `mongoose` for communications between server and mongoDB (link in resources)
- Use .env to store DB configuration (like MONGO_URI) BUT DO NOT PUSH IT TO GITHUB. `.env` file should be shared with the team over slack and is to be maintained locally. 

## How to show its done?
Note that this is just a guideline for creating a proof of concept to show that our DB connection is setup so feel free to play around with it. Take a look at PR from resources for an example

- Create a placeholder model (eg. User) with the following schema
    - User { name: String, email: String }
- Setup a POST route (eg. POST /user) with inputs some arguments for the model you created and stores that model in the DB
- Setup a GET route (eg. GET /user/<id>) which takes ID as input and returns the particular resource with matching ID from DB

Once routes are setup, use POSTMAN to test it and upload a screen recording of the same

## Resources
Intro to MongoDB - https://www.mongodb.com/basics
Mongoose with Typescript - https://mongoosejs.com/docs/typescript.html
Tutorial - https://medium.com/swlh/typescript-with-mongoose-and-node-express-24073d51d2ee
PR from LAKTAA - https://github.com/TritonSE/LAK-Goods-Transport-Application/pull/22

### PR DESCRIPTION
**Pull request by adhi0331**
### Tracking Info

Resolves #6

Make sure your branch name conforms to: `<feature|staging|bugfix|...>/<username>/<3-4 word description separated by dashes>`. Otherwise, please rename your branch and create a new PR.

### Changes

What changes did you make?

We were able to get the app connected to the MongoDB. We also created User models, and routes to get and add users. 

### Testing

How did you confirm your changes worked? 

We tested the routes on postman and were able to get both routes to work properly.

### Confirmation of Change 

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

Changes can be viewed in the app.ts, models folder, and routes folder.

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -14,10 +14,12 @@
   "license": "ISC",
   "dependencies": {
     "dotenv": "^16.0.3",
-    "express": "^4.18.2"
```
Comment (AmanKAggarwal) [inline backend/package.json:17]: Add `body-parser` here. Follow lint instructions for this

File: backend/src/models/users.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { AnyTxtRecord } from 'dns';
+import mongoose from 'mongoose';
+
+interface userInterface{
+    name: String;
```
Comment (AmanKAggarwal) [inline backend/src/models/users.ts:5]: Use `string` not `String`. Check this out for understanding the difference - https://stackoverflow.com/questions/14727044/what-is-the-difference-between-types-string-and-string

**Review by AmanKAggarwal — CHANGES_REQUESTED** [review]
Nice work guys!! PR looks very solid but there are few changes to be made before we can merge
- You might have seen the lint errors on the PR. In case you are not familiar, [linting](https://cogut.medium.com/what-is-linting-how-does-a-linter-work-49381f28fc60) is the process of checking the source code for Programmatic as well as Stylistic errors. Often times it can be annoying and time consuming to detect such errors while reviewing PR, which is why we are using ESLint for this. Make sure you read, understand and fix all the errors the lint check revealed to pass it. 
Use the following command to fix linting errors that can be fixed automatically - 
```
npm run lint-fix
```

- You can ignore the warnings about console.log because in the development phase some console logs are very important eg. Connected to Database
- Update body-parser in package.json. Whenever you add a package, make sure the package.json is being updated so that someone could just `npm install` in our repo and start hacking! 
-  GET /user is not yet complete 

Above are the main things I noticed, I also dropped more specific comments in the PR so address those as well. Overall, these are minor changes so great work integrating mongoDB with our server.

**Comment by AmanKAggarwal** [comment]
> ### Tracking Info
> Resolves #
> 
> Make sure your branch name conforms to: `<feature|staging|bugfix|...>/<username>/<3-4 word description separated by dashes>`. Otherwise, please rename your branch and create a new PR.
> 
> ### Changes
> What changes did you make?
> 
> We were able to get the app connected to the MongoDB. We also created User models, and routes to get and add users.
> 
> ### Testing
> How did you confirm your changes worked?
> 
> We tested the routes on postman and were able to get both routes to work properly.
> 
> ### Confirmation of Change
> Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.
> 
> Changes can be viewed in the app.ts, models folder, and routes folder.

Add `Resolves #6` so that it auto-closes the issue when PR is merged

File: backend/package.json [hunk 3]
```diff
@@ -14,10 +14,12 @@
   "license": "ISC",
```
Comment (YashRavipati1) [inline backend/package.json:14]: Nothing for me to add that Aman didn't address for now. I will review for final approval once the requested changes are made.

**Comment by adhi0331** [comment]
Fixed the ESlint issue and the bug on the get request. Added images below of what the get and post request looks like on my end. 
Screenshot 2023-01-20 at 12 46 45 PM
Screenshot 2023-01-20 at 12 46 25 PM

**Review by AmanKAggarwal — APPROVED** [review]
Great work! Approved for merge

**Review by YashRavipati1 — APPROVED** [review]
Everything looks good

**Pull request merged**